### PR TITLE
DEV: makes chat modals use the new <DModal /> component

### PR DESCRIPTION
**Pull request by jjaffeux**

### CODE REVIEW

File: app/assets/javascripts/select-kit/addon/components/selected-name.hbs [hunk 1]
```diff
@@ -6,6 +6,14 @@
     data-name={{this.name}}
     class="select-kit-selected-name selected-name choice"
   >
+    {{#if this.selectKit.options.formName}}
+      <input
+        type="hidden"
+        name={{this.selectKit.options.formName}}
+        value={{this.value}}
+      />
+    {{/if}}
```
Comment (jjaffeux) [inline app/assets/javascripts/select-kit/addon/components/selected-name.hbs:15]: I added this to be able to use `FormData` in modal/create-channel

File: plugins/chat/assets/javascripts/discourse/components/chat-message-separator-date.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+
+export default class ChatMessageSeparatorDate extends Component {
+  @action
+  onDateClick() {
+    return this.args?.fetchMessagesByDate(
```
Comment (davidtaylorhq) [inline plugins/chat/assets/javascripts/discourse/components/chat-message-separator-date.js:7]: `this.args` will always exist. Did you mean to check for existence of the function?

```suggestion
    return this.args.fetchMessagesByDate?.(
```

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat-message-separator-date.js:7]: yes indeed :thinking:

File: plugins/chat/assets/javascripts/discourse/components/modal/chat/archive-channel.js [hunk 3]
```diff
@@ -95,14 +66,44 @@ export default Component.extend(ModalFunctionality, {
...
+    this.saving = true;
+
+    return this.chatApi
+      .createChannelArchive(this.channel.id, this.#data())
+      .then(() => {
+        this.flash(I18n.t("chat.channel_archive.process_started"), "success");
+        this.channel.status = CHANNEL_STATUSES.archived;
+
+        discourseLater(() => {
+          this.closeModal();
```
Comment (davidtaylorhq) [inline plugins/chat/assets/javascripts/discourse/components/modal/chat/archive-channel.js:90]:
```suggestion
          this.args.closeModal();
```

(`this.closeModal` was previously provided by the `ModalFunctionality` mixin)

File: plugins/chat/assets/javascripts/discourse/components/modal/chat/archive-channel.js [hunk 4]
```diff
@@ -95,14 +66,44 @@ export default Component.extend(ModalFunctionality, {
...
+  }
+
+  @action
+  archiveChannel() {
+    this.saving = true;
+
+    return this.chatApi
+      .createChannelArchive(this.channel.id, this.#data())
+      .then(() => {
+        this.flash(I18n.t("chat.channel_archive.process_started"), "success");
```
Comment (davidtaylorhq) [inline plugins/chat/assets/javascripts/discourse/components/modal/chat/archive-channel.js:86]: Needs to be `this.flash = ` (and we'll need an `@tracked flash` field on the component)

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/modal/chat/archive-channel.js:86]: Yes I guess I forgot that one apparently, will need flashType too as it's success

File: plugins/chat/assets/javascripts/discourse/components/modal/chat/create-channel.hbs [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+              </span>
+              <p class="modal-chat-create-channel__label-description">
+                {{i18n "chat.settings.channel_threading_description"}}
+              </p>
+            </div>
+          </label>
+        </div>
+      {{/if}}
+
+      <div class="modal-footer">
```
Comment (davidtaylorhq) [inline plugins/chat/assets/javascripts/discourse/components/modal/chat/create-channel.hbs:126]: Can now be updated to use https://github.com/discourse/discourse/commit/9a1a3906c4dad3e09b6f0515ccedc78d7bda8bdb

File: plugins/chat/assets/javascripts/discourse/components/chat/modal/archive-channel.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    return this.chatApi
+      .createChannelArchive(this.channel.id, this.#data())
+      .then(() => {
+        this.flash = I18n.t("chat.channel_archive.process_started");
+        this.flashType = "success";
+        this.channel.status = CHANNEL_STATUSES.archived;
+
+        discourseLater(() => {
+          this.ærgs.closeModal();
```
Comment (davidtaylorhq) [inline plugins/chat/assets/javascripts/discourse/components/chat/modal/archive-channel.js:93]:
```suggestion
          this.args.closeModal();
```

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat/modal/archive-channel.js:93]: 😢

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**